ui/src/pages/QuickSetup/SelectRolesPage.tsx

```diff
@@ -185,10 +194,60 @@ const DragAndDropRoles = () => {
   }
 
   function findExistingRole(provider: string, role: RoleKind) {
-    return !!allRoles.find(c => c.role === role && c.contract.payload.provider === provider);
+    return !!allRoles.find(c => c.roleKind === role && c.contract.payload.provider === provider);
   }
 };
 
+const ServiceLabel: React.FC<{ role: Role }> = ({ role }) => {
+  const [hovering, setHovering] = useState(false);
+
+  const ledger = useLedger();
+
+  return (
+    <div
+      className="service-label"
+      onMouseEnter={() => setHovering(true)}
+      onMouseOut={() => setHovering(false)}
+    >
+      {hovering ? (
+        <Label basic onClick={() => terminateRole(role, ledger)}>
+          {role.roleKind} <Icon name="close" />
```

Comment:
Ive been trying to only use our custom icons because they are a bit different than the ones provided by semantic (and we have them anyway) if possible..

Reply:
Also, why not just use `:hover` and `display: none` in scss instead?